src/SprykerEco/Zed/Adyen/Persistence/AdyenRepository.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Adyen\Persistence;
+
+use Generated\Shared\Transfer\PaymentAdyenOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentAdyenTransfer;
+use Generated\Shared\Transfer\SpySalesOrderItemEntityTransfer;
+use Orm\Zed\Adyen\Persistence\SpyPaymentAdyenOrderItemQuery;
+use Orm\Zed\Adyen\Persistence\SpyPaymentAdyenQuery;
+use Orm\Zed\Sales\Persistence\SpySalesOrderItemQuery;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+use Spryker\Zed\PropelOrm\Business\Runtime\ActiveQuery\Criteria;
+
+/**
+ * @method \SprykerEco\Zed\Adyen\Persistence\AdyenPersistenceFactory getFactory()
+ */
+class AdyenRepository extends AbstractRepository implements AdyenRepositoryInterface
+{
+    /**
+     * @param string $reference
+     *
+     * @return \Generated\Shared\Transfer\PaymentAdyenTransfer
+     */
+    public function getPaymentAdyenByReference(string $reference): PaymentAdyenTransfer
+    {
+        $query = $this->getPaymentAdyenQuery()->filterByReference($reference);
+        $entityTransfer = $this->buildQueryFromCriteria($query)->findOne();
+
+        return $this->getFactory()
+            ->createAdyenPersistenceMapper()
+            ->mapEntityTransferToPaymentAdyenTransfer($entityTransfer, new PaymentAdyenTransfer());
```

Comment:
And here - no need for `new PaymentAdyenTransfer()` here and in all the methods below.

Reply:
Answered above.